### PR TITLE
SDL 2.0.10 Updates

### PR DESCRIPTION
**Pull request by TheSpydog**
These are the changes needed to use `SDL_UIKitRunApp` with SDL2-CS. This breaks some backwards compatibility since `SDL_WinRT_mainFunction` is now `SDL_main_func` (which is on SDL, not us), and the `argv` parameter is now just a single IntPtr and not an array.

Think we should merge this now or wait till SDL2.0.10 officially launches?

### CODE REVIEW

**Comment by flibitijibibo** [comment]
If it’s alright with you, let’s turn this into the 2.0.10 PR - we can have all the .10 changes here, and when it finally releases we can just mash the button and go.

**Comment by TheSpydog** [comment]
Fine by me!

**Comment by flibitijibibo** [comment]
It's not been tagged yet, so more stuff might get added, but stuff to review up to this point:

https://hg.libsdl.org/SDL/rev/0f4af9520690

File: src/SDL2.cs [hunk 1]
```diff
@@ -6867,6 +7246,20 @@ public enum SDL_PowerState
...
+
+		/* Only available in SDL 2.0.10 or higher. */
+		[DllImport(nativeLibName, CallingConvention = CallingConvention.Cdecl)]
+		public static extern IntPtr SDL_SIMDAlloc(uint len);
+
+		/* Only available in SDL 2.0.10 or higher. */
+		[DllImport(nativeLibName, CallingConvention = CallingConvention.Cdecl)]
+		public static extern void SDL_SIMDFree(IntPtr ptr);
+		
+		/* Availab
```
Comment (kiates) [inline src/SDL2.cs]: Unterminated comment.

**Pull request merged**